infrastructure/templates/template.json

```diff
@@ -3575,6 +3576,26 @@
         "DeploymentScript": "[parameters('deploymentScript')]"
       }
     },
+    {
+      "condition": "[parameters('deploySubnets')]",
+      "type": "Microsoft.Network/ddosProtectionPlans",
+      "apiVersion": "2020-05-01",
+      "name": "[variables('vNetDdosPlanName')]",
+      "location": "[resourceGroup().location]",
+      "tags": {
+        "Department": "[parameters('departmentName')]",
+        "Solution": "[parameters('solutionName')]",
+        "ServiceType": "Ddos protection plan",
+        "Environment": "[parameters('environmentName')]",
+        "Subscription": "[parameters('subscriptionName')]",
+        "CostCentre": "[parameters('costCentre')]",
+        "ServiceOwner": "[parameters('serviceOwnerName')]",
+        "DateProvisioned": "[parameters('dateProvisioned')]",
+        "CreatedBy": "[parameters('createdBy')]",
+        "DeploymentRepo": "[parameters('deploymentRepo')]",
+        "DeploymentScript": "[parameters('deploymentScript')]"
+      }
+    },
```

Comment:
I was a bit surprised to see that this resource has no properties. Does it not need to be associated with the Vnet or subnets in any way?